src/charm.py

```diff
@@ -18,6 +22,63 @@ class SpringBootCharm(CharmBase):
     def __init__(self, *args):
         """Initialize the instance."""
         super().__init__(*args)
+        self.framework.observe(self.on.config_changed, self.reconciliation)
+
+    def _spring_boot_container(self) -> ops.model.Container:
+        """Retrieve the container for the Spring Boot application.
+
+        Returns:
+            An instance of :class:`ops.charm.Container` represents the Spring Boot container.
+        """
+        container = self.unit.get_container("spring-boot-app")
+        return container
+
+    def _generate_spring_boot_layer(self) -> dict:
+        """Generate Spring Boot service layer for pebble.
+
+        Returns:
+            Spring Boot service layer configuration, in the form of a dict
+        """
+        container = self._spring_boot_container()
+        if not container.isdir("/app"):
+            raise ReconciliationError(new_status=BlockedStatus("dir /app does not exist"))
```

Comment:
Should we capitalize the message ?
Like "Directory /app doesn't exist"

Reply:
Updated, thanks!

Reply:
Let's see the outcome of your pool :)